plugins/modules/monitoring/spectrum_maintenance.py

```diff
@@ -0,0 +1,268 @@
+
+    if info['status'] == 401:
+        module.fail_json(msg="failed to authenticate to Oneclick server")
+
+    if info['status'] not in (200, 201, 204):
+        module.fail_json(msg=info['msg'])
+
+    return response.read()
+
+
+def post(resource, xml=None):
+    return request(resource, xml=xml, method='POST')
+
+
+def put(resource):
+    return request(resource, xml=None, method='PUT')
+
+
+def get_ip():
+    try:
+        device_ip = gethostbyname(module.params.get('device'))
+    except gaierror:
+        module.fail_json(msg="failed to resolve device ip address for '%s'" % module.params.get('device'))
+
+    return device_ip
+
+
+def get_model_handle_by_ip(device_ip):
+    """Query Spectrum for the model_handle of an IP Address"""
+    resource = '/models'
+    # landscape_min = "0x%x" % int(module.params.get('landscape'), 16)
+    # landscape_max = "0x%x" % (int(module.params.get('landscape'), 16) + 0x100000)
+
+    xml = """<?xml version="1.0" encoding="UTF-8"?>
+        <rs:model-request throttlesize="10000"
+        xmlns:rs="http://www.ca.com/spectrum/restful/schema/request"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.ca.com/spectrum/restful/schema/request ../../../xsd/Request.xsd ">
+
+        <rs:target-models>
+            <rs:models-search>
+            <rs:search-criteria xmlns="http://www.ca.com/spectrum/restful/schema/filter">
+                <action-models>
+                    <filtered-models>
+                        <equals>
+                            <model-type>SearchManager</model-type>
+                        </equals>
+                    </filtered-models>
+                    <action>FIND_DEV_MODELS_BY_IP</action>
+                        <attribute id="AttributeID.NETWORK_ADDRESS">
+                            <value>{search_ip}</value>
+                        </attribute>
+                </action-models>
+            </rs:search-criteria>
+            </rs:models-search>
+        </rs:target-models>
+    <rs:requested-attribute id="0x1295d" />
+    <rs:requested-attribute id="0x12aca" />
+    <rs:requested-attribute id="0x12d8b" />
+</rs:model-request> """.format(search_ip=device_ip)
+
+    result = post(resource, xml=xml)
+
+    root = ET.fromstring(result)
+
+    if root.get('total-models') == '0':
+        return None
+
+    namespace = dict(ca='http://www.ca.com/spectrum/restful/schema/response')
+
+    # get the first device
+    model = root.find('ca:model-responses', namespace).find('ca:model', namespace)
+
+    if model.get('error'):
+        module.fail_json(msg="error checking device: %s" % model.get('error'))
+
+    # get the attributes
+    model_handle = model.get('mh')
+    is_managed = model.find('./*[@id="0x1295d"]').text
+    is_not_hibernating = model.find('./*[@id="0x12aca"]').text
+    hibernating_after_maintenance = model.find('./*[@id="0x12d8b"]').text
```

Comment:
Can the text be further processed by a function which converts `'false'` to `False` and `'true'` to `True`?
Deferring the string comparisons further down in `main` is a little confusing.